### ISSUE TITLE
Not all tests run in test_groupby.py

### ISSUE THREAD

**Issue by joergdietrich** [dask/dask#3443]
Unless I'm missing something, there are three methods in test_groupby.py that look like tests but are not run by pytest because they are not prefixed by `test_`:

```
$ pytest --collect-only dask/dataframe/tests/test_groupby.py | egrep '(groupby_internal_repr|groupby_error|groupby_internal_head)'
  <Function 'test_series_groupby_errors'>
```

The first two of these would fail with a `ValueError` because

```
    pdf = pd.DataFrame({'x': [1, 2, 3, 4, 6, 7, 8, 9, 10],
                        'y': list('abcbabbcda')})
```

`x` and `y` have different lengths (further test failures may of course happen if this is fixed).

I'd be happy to submit a PR. I'm just not sure I'm not missing something here.

**Comment by mrocklin** [comment]
Git blame shows that @sinhrks wrote these.  Perhaps he has some insight
here?

From my perspective they should be run, and so yes, I would recommend
prepending test_ to their names.

**Comment by jrbourbeau** [comment]
Closed by #3480
